ui/app/components/pending-tx.js

```diff
@@ -342,22 +343,22 @@ PendingTx.prototype.miniAccountPanelForRecipient = function () {
   }
 }
 
-PendingTx.prototype.gasPriceChanged = function (newHex) {
-  log.info(`Gas price changed to: ${newHex}`)
-  const inWei = hexToBn(newHex).mul(GWEI_FACTOR)
+PendingTx.prototype.gasPriceChanged = function (newBN) {
+  log.info(`Gas price changed to: ${newBN.toString(10)}`)
   const txMeta = this.gatherTxMeta()
-  txMeta.txParams.gasPrice = inWei.toString(16)
-  this.setState({ txData: txMeta })
+  txMeta.txParams.gasPrice = '0x' + newBN.toString('hex')
+  this.setState({ txData: cloneObj(txMeta) })
```

Comment:
why are you cloning the txMeta? were there some side effects? (if so, maybe we should do this everywhere we use `setState`)